### DISCUSSION TITLE
Working on Akai Ax80 panel: how to add sysex send receive?

### DISCUSSION THREAD

**Discussion by Miditribe** [RomanKubiak/ctrlr#645] [Q&A]
Hi all, I'm working (final phase) on an Akai Ax80 panel, for all models with Tauntek Os upgraded. I need to add sysex send receive. I'm not a programmer just working on panels, ideas, cc and a bit of sysex msgs. I've found the syx messages list for Ax80 but I can't find a way to send or receive sysex from or to a specific patch number (Ax has A1-32 and B1-32 plus 32 rom presets). What I need is a way to store patches after editing. Please help. 
Lu

**Comment by damiensellier** [comment]
Are you just looking to request sysex patch dumps from the Akai?

**Reply by damiensellier** [reply]
The logic would be:

Send a sysex dump request to the Akai with the program variable inserted into the message.

when the reply comes back from the Akai to the ctrlr/panel, filter this specific type with message size and message type with the global method "when the panel receives a message". If the type is a valid program sysex dump, then store it as Memoryblock and do what you have to do from there (save as file, decypher and update the panel knobs etc)

![Capture d’écran, le 2024-05-27 à 16 51 49](https://github.com/RomanKubiak/ctrlr/assets/55164385/4deb3205-4ae7-4f31-9d78-07cb68663f1e)


This is a very common method for sysex presets dump.

All depends on what you want to do with it.

**Reply by damiensellier** [reply]
@dnaldoog, did a generic sysex dump panel with custom requests if it is too intense for you:

![GSDR_ss](https://github.com/RomanKubiak/ctrlr/assets/55164385/f8115f6c-dec6-4aa2-9f21-5e80c68da71d)

https://ctrlr.org/generic-sysex-dump-editor/

The code from that panel would be the perfect place to start if you want to implement it in your Akai panel.

**Comment by Miditribe** [comment]
@damiensellier thanks for the quick responses. I think that even with the upgraded os from Tauntek you have to manually start dump saving patches, but no need to push buttons if receiving syx. I just need to save syx files, one or all patches at once, and then reload, one or all. All 45 Ax80 editing parameters are done now so this is the last step before publish.  About the upgrade:
[AX80NewFWinstrs.pdf](https://github.com/RomanKubiak/ctrlr/files/15462228/AX80NewFWinstrs.pdf)

**Reply by damiensellier** [reply]
Oh okay I see, there's no sysex message you can send to the AKAI to request the program dumps.
So you just need the generic panel I posted above to record them as files on your computer and send them back to the machine.

I used Sysex Librarian by Snoize for storing sysex patches. It's for macOS : https://www.snoize.com/SysExLibrarian/

**Comment by Miditribe** [comment]
Thanks. One last chance is to edit the panel, then save it as a syx and then send it to the Ax80 without any fuss, since it will receive without touching any button or making a request. But here more problems arise because I use CC msgs not sysex. 
Few words from Bob Grieg (new Os creator):

"SYSEX Patch save and restore: 
It is now possible to save and restore all 64 Bank A and B patches using MIDI. To save, press Tape, then 
30 (Save). If you press Save a second time, the patches will be sent to the cassette interface, as before. 
**_Reception does not require any special button pushes._** The patches are sent as 64 complete units, each with a 
header, a program number and the sysex end byte. The format is as follows: 
F0 47 7E pp dd dd dd … F7 Since the program number (0-63) and all parameter values fit in seven bits, 
there was no need to split the data into low and high nibbles for MIDI. The dd bytes contain the parameter 
values for a patch, starting with 1, then 2, etc. up to parameter 45".**_

**Reply by damiensellier** [reply]
It would require to then form a full sysex message from your panel parameter settings before pushing them to the AKAI when needed.
It is doable for sure. 
pp for program number? dd for parameter value I guess.
Can you please save a preset sysex dump and see if the message relates to your panel settings? You can paste the received message here to decypher.
We will see what length is the message and see if all parameter values are there.
Do you have the list of CC messages as well to check if their values are at their respective position of the sysex message?

**Comment by Miditribe** [comment]
Thanks a lot, here you go: there are 45 parameters on sysex msg. - other than the F7 and the Akai model - same as the parameters on the Ax80. As you know, apart the first three hex couples (F0 47 7E) all is related to the parameters. 
pp = 0-99 (not all there are exceptions, for freq. range for example are 0-1-2, all explained in the attached pdf).
This s is the first patch: 
F0 47 7E 00 01 01 63 04 01 63 23 33 01 00 31 01 20 30 08 35 5A 63 1A 00 00 00 03 00 0A 24 00 0C 5A 00 63 63 00 11 00 02 00 00 00 03 0E 17 0B 63 00 F7
There are 2 Ram banks of 32 patches (A 1-32 and B1-32) and a preset bank (P1-32)
I made a document with all cc messages needed. Note that there are 32 buttons and 45 param. 
Is it possible to assign a single slider to each parameter and then send a single syx to the synth?
![20240522_150319](https://github.com/RomanKubiak/ctrlr/assets/158320459/d3847207-688e-4516-bc4d-aeff8767c359)
![20240522_150333](https://github.com/RomanKubiak/ctrlr/assets/158320459/539824bc-50bc-4e90-adbd-be2240b1a00a)
![20240522_150336](https://github.com/RomanKubiak/ctrlr/assets/158320459/5b62bb53-c8df-42eb-a973-75bdff6cf61e)
![20240522_150344](https://github.com/RomanKubiak/ctrlr/assets/158320459/e48f92af-3269-4d60-a4d8-1e4dc134b02a)
[AKAI AX 80 CC MESSAGES EDIT.odt](https://github.com/RomanKubiak/ctrlr/files/15481938/AKAI.AX.80.CC.MESSAGES.EDIT.odt)

**Comment by dnaldoog** [comment]
Does this panel work? Just for two controls. 🤔

[Akai AX80_1_0.zip](https://github.com/RomanKubiak/ctrlr/files/15483835/Akai.AX80_1_0.zip)


![image](https://github.com/RomanKubiak/ctrlr/assets/76831683/d32b0f09-86b3-4a3f-b4a0-d0eeaaf40fe7)

**Reply by Miditribe** [reply]
Yes! Great @dnaldoog... they work. If you can, could you explain what's in there since I tried the same way but got no editing at all? it's a matter of xx only (and hex parameter) in sysex string?Two  more questions: 
1) now I can do the same on the panel but is there a way to add the other hex to the sysex of each slider? To automate and speed up panel editing, creating multiple sliders at once with progressive hex.
2) the last problem is: how to collect all these infos about the patch and send them as a syx file to the Ax80?
PS: Obviously you are in the project if this is ok for you. Same for @damiensellier

**Reply by damiensellier** [reply]
If johns formula works, you would just need to lay down the 45 parameters sliders with index 0 to 44 then assign on mod value change a LUA method :
F0 47 7E 40 index value F7 
and you'd get control over the parameters with sysex..
If you want to save the current panel as sysex dump you would need to create a table of all parameters as index and their respective value and create a LUA formula placing the value of all parameters at their byte position/index .
This is what you want to achieve?

This synth is a beauty!

**Reply by Miditribe** [reply]
Yes, will try it. I'm gonna lay down 45 sliders with progressive index and then assign on each mod value change (where?) a LUA method
Then I would save all 45 parameters as a syx string to send it to Ax80: yes - create a table and send so the patch is saved once editing has been done- but how? Sorry, my knowledge of Ctrlr is bad
Yes This synth is a Eighties beauty!

**Reply by damiensellier** [reply]
```
sendParamValueSysex = function(--[[ CtrlrModulator --]] mod, --[[ number --]] value, --[[ number --]] source)

modIndex = mod:getProperty("vstIndex")
console("modIndex: "..modIndex)

paramValueSysexMsg = CtrlrMidiMessage(MemoryBlock({0xF0, 0x47, 0x7E, 0x40, tonumber(modIndex), value, 0xF7}))
panel:sendMidiMessageNow(paramValueSysexMsg)

end
```

**Reply by damiensellier** [reply]
Yep, I have it working:

<img width="1005" alt="Capture d’écran 2024-05-29 à 12 00 22" src="https://github.com/RomanKubiak/ctrlr/assets/55164385/9c09f1dd-7e4e-4bfc-be32-0017446f4bec">

[AX80_1_0_2024-05-29_11-20.panel.zip](https://github.com/RomanKubiak/ctrlr/files/15487194/AX80_1_0_2024-05-29_11-20.panel.zip)

I did it with CtrlrX 5.6.30 with the YamDX template, they are the same colourScheme as the Yamaha DX7

I assigned the Index number to the vstIndex parameter then used it as Parameter Index in the formula and converted it to number since it's seen as a string.

**Reply by damiensellier** [reply]
That was to send the mod value as sysex.

Now we need to cast the sysex program by iterating through all the mods by index 0 to 44 and place their respective value in a table. 
Once in a table we need to run a loop and get a sysex message with all 45 values in a raw. I'll try it tonight on my free time.

**Reply by damiensellier** [reply]
Can you please check if the program request message would be working : 
0xF0, 0x47, 0x7E, 0x00, 0xF7 (for program 00)
or 
0xF0, 0x47, 0x7E, 0x01, 0xF7 (for program 01)

If you have the program sysex as a reply you're all set to be able to display all the parameters of your synth in the panel in real time without having to press the buttons to request the dump.

**Reply by Miditribe** [reply]
I can't find ctrlr 5.6.30, I'm on 6.0.4 and it crashes

**Reply by damiensellier** [reply]
https://github.com/RomanKubiak/ctrlr/discussions/619#discussioncomment-8779822

**Reply by damiensellier** [reply]
It's working on mac even with 5.3.198, I just checked, just the GUI is a bit different.
Are you on PC?

**Reply by damiensellier** [reply]
Your crash is probably coming from the sendMidiMessageNow() discrepancy between old and newer versions.
Try this one:

[AX80_LEGACY_1_1_2024-05-29_14-24.panel.zip](https://github.com/RomanKubiak/ctrlr/files/15489245/AX80_LEGACY_1_1_2024-05-29_14-24.panel.zip)

**Comment by damiensellier** [comment]
It was a long time I did not work on some memoryBlock conversion, that was a good exercise to train myself. Here is the script I did to cast the full program sysex message: 

1. I go through all 45 modulators from their vstIndex number and get their value
2. those values are stored in a table
3. the table is converted to a memoryBlock
4. I concatenate the message with the sysex prefix and the suffix
5. the message is sent

```
--
-- Called when a modulator value changes
-- @mod   http://ctrlr.org/api/class_ctrlr_modulator.html
-- @value    new numeric value of the modulator
--
sendSnapshotSysexPrgm = function(--[[ CtrlrModulator --]] mod, --[[ number --]] value, --[[ number --]] source)

    local modValueTable = {}
    for i = 0, 44 do
        local modVstIndex = panel:getModulatorWithProperty("vstIndex",i)
        local modValue = modVstIndex:getModulatorValue()

        modValueTable[i+1] = modValue
        --console(tostring(modValue))
    end

    local PrgmDest = 0 -- Program Destination to be declared
    local PrgmSysexMsg = MemoryBlock(string.format("f0 47 7e %02X", PrgmDest))
    local EOX = MemoryBlock("F7") -- End of SySex Message is always F7

    local snapshotMb = MemoryBlock()
    snapshotMb:createFromTable(modValueTable)
    --console("Displays snapshotMb")
    --console(String(snapshotMb:toHexString(1)))

    PrgmSysexMsg:append(snapshotMb)
    PrgmSysexMsg:append(EOX)

    console("Displays The full PrgmSysexMsg")
    console(String(PrgmSysexMsg:toHexString(1)))


    PrgmSysexMsg = CtrlrMidiMessage(PrgmSysexMsg)
    panel:sendMidiMessageNow(PrgmSysexMsg)

end
```

you'll need to find a way to replace the program destination with a popup, a field a drop down or whatever you want in order to feed the PrgrmDest variable in the sysex formula

Hope it helps

Damien

[AX80_1_4_2024-05-29_23-36.panel.zip](https://github.com/RomanKubiak/ctrlr/files/15493351/AX80_1_4_2024-05-29_23-36.panel.zip)

EDIT: I updated the panel since the message intro was wrong.

**Comment by dnaldoog** [comment]
This panel adds to Damien's panel. It should do everything you need and work in 5.3.198 or 5.6

[AX80_2_0.zip](https://github.com/RomanKubiak/ctrlr/files/15493348/AX80_2_0.zip)

**Reply by damiensellier** [reply]
I changed the intro of the message, it was not the same as the one for parameter values ;)

**Reply by damiensellier** [reply]
> This panel adds to Damien's panel. It should do everything you need and work in 5.3.298 or 5.6
> 
> [AX80_2_0.zip](https://github.com/RomanKubiak/ctrlr/files/15493348/AX80_2_0.zip)

You've been way shorter than me about the way to cast the message! bravo :)

**Reply by dnaldoog** [reply]
> I changed the intro of the message, it was not the same as the one for parameter values ;)

Great - I did it at a fair clip so no debugging, but I think that should cover the basics he needs :)

**Reply by Miditribe** [reply]
First thanks for your help guys. I'm on pc win 11. Gonna try later when I'm back home. Glad to see that a mountain to climb for me is an exercise for you!

**Reply by Miditribe** [reply]
Sorry but no panel works. The parameters editing is ok, save and load also. Nothing happens after sending syx.

**Reply by dnaldoog** [reply]
So when you change an individual parameter it works and changes a value on the AX80?

Is the panel **sending** a bulk dump? Can you see on a MIDI monitor?

It should be. And is the syntax correct? I have no way of knowing whether there is some small bug somewhere (_there usually is_). It would be rare to write a panel for a synth one doesn't have and have it work. You'll need to do some testing.

**Reply by Miditribe** [reply]
Ok I've found that the saved syx from your panel has 49 hex not 50 as the original one

**Reply by Miditribe** [reply]
F0 47 7E 14 36 65 2A 31 21 0E 18 44 31 1C 33 01 31 4B 2D 67 3B 44 34 5C 4E 12 00 55 5E 73 6A 42 3B 6F 42 60 3D 31 4B 47 21 3B 34 44 40 4D 73 28 F7 (49 total)
F0 47 7E 00 01 01 63 04 01 63 23 33 01 00 31 01 20 30 08 35 5A 63 1A 00 00 00 03 00 0A 24 00 0C 5A 00 63 63 00 11 00 02 00 00 00 03 0E 17 0B 63 00 F7 (50)

**Reply by Miditribe** [reply]
So it's a one parameter issue. Gonna check it

**Reply by Miditribe** [reply]
Confirmed: if I add a hex (49 plus one added) it works

**Comment by dnaldoog** [comment]
Excellent news!

**Reply by Miditribe** [reply]
yes, but, where the error is? Maybe one slider not connected to the syx trick?

**Reply by damiensellier** [reply]
Can you please post the dump of program 00, program 01 and program 63 ? thanks

**Reply by Miditribe** [reply]
The dump saved from your panel, or the original dump from the Ax80?

**Reply by damiensellier** [reply]
forget about it. I found the problem. **The program destination pp byte is missing** in @dnaldoog panel. hence 49 bytes vs 50.
Mine is sending 50 bytes but sends only program 00 since it's hard coded in the fomula. just change the following code in my lua script :
`    local PrgmDest = 0 -- Program Destination to be declared`
then "save and compile" 

it should be working for whatever program you change. 

[AX80_1_4_2024-05-29_23-36.panel.zip](https://github.com/RomanKubiak/ctrlr/files/15493351/AX80_1_4_2024-05-29_23-36.panel.zip)

**Reply by damiensellier** [reply]
And don't forget to assign your midi output to the right one of course.
let us know, we'll start from there.

**Reply by Miditribe** [reply]
Ok Will check later, thanks a lot, I'll let you know

**Reply by Miditribe** [reply]
Ok after few hours of work I can tell that @damiensellier last "send" works. Save and load file too (this is weird because of the 49 bytes vs original 50 bytes needed). But at the end you have to save patches directly on the AX80. But is a good starting point! The perfect solution should be a mix of @damiensellier and @dnaldoog ways to wok with syx.. A merge of "send" syx and load plus save. Next duty is (for me) to number and name all sliders, give them some candy graphics. Is there a way to post the panel here? Or send it via message? Thanks

**Reply by damiensellier** [reply]
There you go, this is a mix of both panels. 
You have a dialog window asking for the destination program to your AKAI. 

![Capture d’écran, le 2024-06-01 à 15 10 00](https://github.com/RomanKubiak/ctrlr/assets/55164385/5771550a-3ffd-4de1-af45-d0338edec7da)



Now it's on you ;)

[AX80_2_1_2024-06-01_14-10.panel.zip](https://github.com/user-attachments/files/15523001/AX80_2_1_2024-06-01_14-10.panel.zip)

**Reply by Miditribe** [reply]
Fantastico! More info tomorrow

**Reply by Miditribe** [reply]
Back to you guys. Ok It works but we have another issue: the saved file or the bulk data (I suspect) are a carbon copy of hex's on the panel. This means that every parameter is copied and saved or sent. This also means that I have to change all parameters to have the real sound I need because patch on the keyboard is not loaded, is only edited. Example: if I edit resonance only, saving or sending  it will copy the panel sliders in the state they are. Hope to be clear. Another 2 little things: @damiensellier, hope not to bother you, can we save to parameters A1-A32 and B1-B32, instead of 0-63? Is that possibile? And, this I will do myself: can I add cc parameters to sysex sliders? Just because I made a Nektar Panorama P1 adaptation and it works on cc only. Thanks

**Reply by damiensellier** [reply]
You mean the patch is not automatically applied ?
Maybe you'll just need a program change with the corresponding program following the sysex message to update the keyboard parameters?
This is when not having the hardware at hand is almost impossible to work on.


You would need 2 drop down in the popup :  bank A-B and programs 00-31, that's it?

For the CC part you just need to edit the knobs one by one with their name and the cc number with the appropriate message, don't forget to select the message type to CC in the drop down selector. Also reset the "when the modulator value changes" to **--none** since you'll get a CC message sent to avoid duplicates.

**Reply by Miditribe** [reply]
The patch do not reflect the actual sound but the sliders status ( I suppose). And about the patches, they are A 01-32 and B 01-32. This will simplify the editing and saving, since on the Ax80 you have patches numbered the same.

**Reply by damiensellier** [reply]
if the patch do not reflect the actual sound, it makes no sense.
Maybe the different sliders are not calibrated properly. 
are they all going from the 0 to 127 range? or some goes negative/positive etc? That could be the reason the patches are not loading properly.

**Reply by damiensellier** [reply]
```
For parameter 1, a value of 0 selects 16’, 1 is 8’ and 2 is 4’.
For parameter 7, the mapping is:
0-11: 16/0-11, 12-23:8/0-11, 24-35:4/0-11, 36-47:2/0-11 So they are showing the interval between
DCO2 and DCO1 as an octave and a semitone
```

It means you have to work with uiFixedSliders for those parameters with the appropriate correspondance (i.e : 16=0,8=1,4=2)

https://www.vintagesynth.com/sites/default/files/2017-05/Akai_AX80_Owners_Manual.pdf the list of param are on page 11.
I think param 0 (OSC 1 FREQ. RANGE is CC0), you'd first need to change the different modulators visible name that would be a nice step. then replace the non linear listed above by "uiFixedSliders". Just right click on the one you want to replace the type with.

![replace-with-fixedslider](https://github.com/RomanKubiak/ctrlr/assets/55164385/f25014f3-e5ef-441d-8337-ec83ee83a344)


And for non linear (0-127) just add the number of steps in the panel midi section and the corresponding value like this : 

![Capture d’écran, le 2024-06-03 à 21 43 06](https://github.com/RomanKubiak/ctrlr/assets/55164385/fa265e4b-f4ac-4799-9016-54673f76b054)

![Capture d’écran, le 2024-06-03 à 21 49 53](https://github.com/RomanKubiak/ctrlr/assets/55164385/4f3d673e-c01c-4b60-b335-9a996efa2725)

![Capture d’écran, le 2024-06-03 à 21 49 46](https://github.com/RomanKubiak/ctrlr/assets/55164385/7f69db2d-7bdd-4fa1-b43b-ca875e8669cf)

Let me know when you have everything set properly with the right visible names, sliders' types and corresponding values.
Good luck!

**Reply by damiensellier** [reply]
NOTE : In the manual, this is your uiSliderMin and uiSliderMax values (0-99) for exemple
![Capture d’écran, le 2024-06-03 à 22 04 20](https://github.com/RomanKubiak/ctrlr/assets/55164385/b7063396-ba97-4a60-a3ce-1fcb2a0bae48)

**Reply by Miditribe** [reply]
Yes, 0-99 is the range. Cc: I will add to sliders and see... Thank you so much

**Reply by Miditribe** [reply]
Here is it. Not eye candy but 80% working (Load save and send not)
[Uploading AX80 ALPHA 0.zip…]()

**Reply by damiensellier** [reply]
Sorry but your upload failed.

**Reply by Miditribe** [reply]
[Uploading AX80 ALPHA 0.zip…]()

**Reply by damiensellier** [reply]
Sorry but your upload failed again. Give it some time until you hit the "Reply" button.

**Reply by Miditribe** [reply]
Ok let's try again
[AX80 ALPHA 0.zip](https://github.com/user-attachments/files/15610846/AX80.ALPHA.0.zip)

**Reply by Miditribe** [reply]
Hey, I saw this here: https://github.com/RomanKubiak/ctrlr/discussions/627 
are there any news about? Would be very nice on Ax80

**Reply by damiensellier** [reply]
I thought you would post it as a panel file not an exe. I'm on mac :)

concerning the LinearBarVertical you can already define this type to modulator with the following line in the onLoading script:
`panel:getSlider("slider_name"):getOwnedSlider():setSliderStyle(Slider.LinearBarVertical)`

**Reply by Miditribe** [reply]
Ok. And I will post the panel

**Reply by Miditribe** [reply]
[ax80 alpha.zip](https://github.com/user-attachments/files/15722010/ax80.alpha.zip)
panel panelz

**Reply by Miditribe** [reply]
?

**Reply by damiensellier** [reply]
I don't know why, but when I touch sliders on your panel, it crashes my ctrlr app. It wasn't crashing before. But you seem to have changed things in the scripts, maybe it comes from that.
So I recommand you go all the way in your direction and when you reach the 99% done mark, just give us a shout on what you would need help with.
Take care and good luck with your project!
Damien

**Reply by Miditribe** [reply]
Here is it. Is a cc messages job only but complete with almost anything on the Ax80. Only need syx send receive save 
[ax80 alpha OK.zip](https://github.com/user-attachments/files/15787211/ax80.alpha.OK.zip)

**Reply by damiensellier** [reply]
It's not crashing on my side now. You redesigned the panel from scratch?

**Reply by damiensellier** [reply]
If you want to insert the script for sysex messages we worked on with John, you would need to rename the modulators approprately in the order to the given **button number** in the manual, now the modulator name and vstindex are messed up:

![Capture d’écran, le 2024-06-11 à 16 33 18](https://github.com/RomanKubiak/ctrlr/assets/55164385/6ce13ea9-4838-49fc-a155-9bb184e99384)

You can use the **Panel>Modulator List**   tool to go faster.

**Reply by damiensellier** [reply]
The script mapping the value from/to the modulators works by iterating through the modulator name (not display name) : 

```
for i = 0, 44 do
   local val = panel:getComponent(string.format("EDIT%.2d", i)):getValue()
   table.insert(t, val)
end
```

So they have to be in the right order with the right modulator name.
You can use whatever prefix you want instead of EDIT but the following value has to be the matching one to the button number in the AX80 manual.

**Reply by Miditribe** [reply]
Ok thank you I will try. I made this panel just before your last attempt but I did not tried it throughly.

**Reply by Miditribe** [reply]
I changed modulators number. But, if I copy and paste "load save bulk" on the last panel commands do not work at all. Copy and paste is not the way to add these buttons, is it?
![Untitled 55](https://github.com/RomanKubiak/ctrlr/assets/158320459/0bb40290-30ad-4467-a021-bec65634b2b2)

**Reply by damiensellier** [reply]
you have to assign LUA scripts to those buttons

**Reply by Miditribe** [reply]
just copied my panel into yours... 80% ok but no load send and bulk (window opens and stay opened). Graphics are incomplete/messed up
Short to do list:
1) bulk dump of all 64 patches at once        
2) save load starting from patch number A 1-32 and  B 1-32 (bank selectable)
3) Vertical sliders a la Ax80 displays
consider that the only working editor librarian for Ax80 is inside Sound Quest (hundred bucks). 
so if you are interested we can try to sell for cheap or via donations!
thanks for your efforts guys
[AX80 FINAL ALPHA SEND LOAD BULK NOT WORKING.zip](https://github.com/user-attachments/files/15823009/AX80.FINAL.ALPHA.SEND.LOAD.BULK.NOT.WORKING.zip)

**Reply by damiensellier** [reply]
in the LUA scripts you have a loop going through all modulators names "EDIT" + i (with i being the iterator): 

```
        for i = 0, 44 do
            local val = panel:getComponent(string.format("EDIT%.2d", i)):getValue()
            table.insert(t, val)
        end
```

But in you modulator list there's no EDIT01, EDIT02 etc  you have modulator-01 etc

![Capture d’écran, le 2024-06-13 à 13 44 17](https://github.com/RomanKubiak/ctrlr/assets/55164385/71da60b2-0ed2-4c40-8311-dca3426430fd)


You would need to organize your modulator names accordingly. so either you rename your modulator EDITsomething or you change the loop formula. And as mentionned before, your modulator name (ie: EDIT01, EDIT02...) have to be in the same order as the list of parameters in the manual to place the parameter values in the sysex message at the right place.

Good luck
